fcrepo-http-api/src/test/java/org/fcrepo/integration/http/api/FedoraLdpIT.java

```diff
@@ -979,9 +980,38 @@ public void testCreateGraph() throws IOException {
         }
     }
 
-
     @Test
     public void testCreateVersionedRDFResource() throws IOException {
+        createVersionedRDFResource();
+    }
+
+    @Test
+    public void testGetVersionedResourceHeaders() throws IOException {
+        final String subjectURI = createVersionedRDFResource();
+        try (final CloseableHttpResponse response = execute(new HttpGet(subjectURI))) {
+            verifyVersionedResourceResponseHeaders(subjectURI, response);
+        }
+    }
+
+    @Test
+    public void testHeadVersionedResourceHeaders() throws IOException {
+        final String subjectURI = createVersionedRDFResource();
+        try (final CloseableHttpResponse response = execute(new HttpHead(subjectURI))) {
+            verifyVersionedResourceResponseHeaders(subjectURI, response);
+        }
+    }
+
+    private void verifyVersionedResourceResponseHeaders(final String subjectURI,
+            final CloseableHttpResponse response) {
+        assertEquals("Didn't get a CREATED response!", OK.getStatusCode(), getStatus(response));
+        checkForVersionedResourceLinkHeader(response);
+        checkForLinkHeader(response, subjectURI, "timegate");
+        checkForLinkHeader(response, subjectURI + "/TimeMap", "timemap");
```

Comment:
The actual URI for timemap (and memento) resources will _not_ include the `/TimeMap` node name, but instead will be served through `/fcr:versions` endpoint on the resource.
We can leave this for this PR, but I wanted to raise your attention to the design... and associated ticket: https://jira.duraspace.org/browse/FCREPO-2646

Reply:
good to know.